### PR TITLE
Feature/android fix serial no

### PR DESCRIPTION
**Pull request by Dartlexx**

### CODE REVIEW

File: vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/AndroidDevice.kt [hunk 1]
```diff
@@ -37,7 +37,7 @@ import java.util.*
 import kotlin.coroutines.CoroutineContext
 
 class AndroidDevice(val ddmsDevice: IDevice,
-                    private val serialStrategy: SerialStrategy = SerialStrategy.AUTOMATIC) : Device, CoroutineScope {
+                    private val serialStrategy: SerialStrategy = SerialStrategy.DDMS) : Device, CoroutineScope {
```
Comment (Malinskiy) [inline vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/AndroidDevice.kt:40]: This is settable by Marathonfile. Don't see any reason to change the sourcecode for this - just use the Marathonfile

File: vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/AndroidDevice.kt [hunk 2]
```diff
@@ -83,20 +83,17 @@ class AndroidDevice(val ddmsDevice: IDevice,
...
      */
     private val realSerialNumber: String by lazy {
-        val marathonSerialProp: String = ddmsDevice.getProperty("marathon.serialno") ?: ""
         val serialProp: String = ddmsDevice.getProperty("ro.boot.serialno") ?: ""
         val hostName: String = ddmsDevice.getProperty("net.hostname") ?: ""
         val serialNumber = ddmsDevice.serialNumber
 
         val result = when (serialStrategy) {
             SerialStrategy.AUTOMATIC -> {
-                marathonSerialProp.takeIf { it.isNotEmpty() }
```
Comment (Malinskiy) [inline vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/AndroidDevice.kt:93]: See https://github.com/Malinskiy/marathon/issues/223#issuecomment-491481014 for why this should be here. Marathon serial property was implemented for agoda's device farm

File: vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/AndroidDevice.kt (hunk 1 above)
Comment (Dartlexx) [reply vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/AndroidDevice.kt:40]: Thanks for the input, but as far as I know there is no such parameter in that version of the Marathon project. Is it present in the latest version of the main project?

File: vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/AndroidDevice.kt (hunk 2 above)
Comment (Dartlexx) [reply vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/AndroidDevice.kt:93]: Well, there is no such property on emulators - most probably this feature is broken in adb-butler. So I needed a quick fix for new emulators, that don't have any other id.

**Comment by Dartlexx** [comment]
@Malinskiy Do you think it is worth it to try and merge the latest changes from main project in your repo?

File: vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/AndroidDevice.kt (hunk 2 above)
Comment (Malinskiy) [reply vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/AndroidDevice.kt:93]: There was one: https://github.com/agoda-com/adb-butler/commit/1171a5f459d84e958f9f3c9b195208b5ee89e929

**Comment by Malinskiy** [comment]
We're still working on marathon and support it. It would be wise to sync the changes from time to time.

File: vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/AndroidDevice.kt (hunk 1 above)
Comment (Dartlexx) [reply vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/AndroidDevice.kt:40]: @Malinskiy I was wrong. I will rollback this PR

**Pull request closed without merging**